mne_connectivity/spectral/time.py

```diff
@@ -784,51 +843,62 @@ def _spectral_connectivity(
+    # check that spectral mode is recognised
+    _check_option("mode", mode, ("cwt_morlet", "multitaper"))
+
+    # compute time-frequency decomposition
+    mt_bandwidth = mt_bandwidth if mt_bandwidth else 4
+    if not spectrum_computed:
+        data = np.expand_dims(data, axis=0)
+        kw_cwt.setdefault("zero_mean", False)  # avoid FutureWarning
+        if mode == "cwt_morlet":
+            out = tfr_array_morlet(
+                data,
+                sfreq,
+                freqs,
+                n_cycles=n_cycles,
+                output="complex",
+                decim=decim,
+                n_jobs=n_jobs,
+                **kw_cwt,
+            )
+        else:
+            out = tfr_array_multitaper(
+                data,
+                sfreq,
+                freqs,
+                n_cycles=n_cycles,
+                time_bandwidth=mt_bandwidth,
+                output="complex",
+                decim=decim,
+                n_jobs=n_jobs,
+                **kw_mt,
+            )
+        out = np.squeeze(out, axis=0)
+    else:
+        out = data
+
+    # give tapers dim to cwt_morlet output
     if mode == "cwt_morlet":
-        out = tfr_array_morlet(
-            data,
-            sfreq,
-            freqs,
-            n_cycles=n_cycles,
-            output="complex",
-            decim=decim,
-            n_jobs=n_jobs,
-            **kw_cwt,
-        )
-        out = np.expand_dims(out, axis=2)  # same dims with multitaper
-        weights = None
-    elif mode == "multitaper":
-        out = tfr_array_multitaper(
-            data,
-            sfreq,
-            freqs,
-            n_cycles=n_cycles,
-            time_bandwidth=mt_bandwidth,
-            output="complex",
-            decim=decim,
-            n_jobs=n_jobs,
-            **kw_mt,
-        )
+        out = np.expand_dims(out, axis=1)
+
+    # compute taper weights
+    if mode == "multitaper":
         if isinstance(n_cycles, int | float):
             n_cycles = [n_cycles] * len(freqs)
-        mt_bandwidth = mt_bandwidth if mt_bandwidth else 4
-        n_tapers = int(np.floor(mt_bandwidth - 1))
-        weights = np.zeros((n_tapers, len(freqs), out.shape[-1]))
+        n_tapers = out.shape[-3]
+        n_times = out.shape[-1]
+        half_nbw = mt_bandwidth / 2.0
+        weights = np.zeros((n_tapers, len(freqs), n_times))
         for i, (f, n_c) in enumerate(zip(freqs, n_cycles)):
             window_length = np.arange(0.0, n_c / float(f), 1.0 / sfreq).shape[0]
-            half_nbw = mt_bandwidth / 2.0
-            n_tapers = int(np.floor(mt_bandwidth - 1))
             _, eigvals = dpss_windows(window_length, half_nbw, n_tapers, sym=False)
             weights[:, i, :] = np.sqrt(eigvals[:, np.newaxis])
             # weights have shape (n_tapers, n_freqs, n_times)
     else:
-        raise ValueError("Mode must be 'cwt_morlet' or 'multitaper'.")
-
-    out = np.squeeze(out, axis=0)
+        weights = None
```

Comment:
Just some refactoring here to reduce code repetition with the new `EpochsTFR` support.